test/test_fcntl.rs

```diff
@@ -27,6 +30,31 @@ fn test_openat() {
     close(dirfd).unwrap();
 }
 
+#[test]
+fn test_rename() {
+    let old_dir = tempfile::tempdir().unwrap();
+    let old_path = old_dir.path().join("old");
+    File::create(&old_path).unwrap();
+    let new_dir = tempfile::tempdir().unwrap();
+    let new_path = new_dir.path().join("new");
+    rename(&old_path, &new_path).unwrap();
+    assert_eq!(rename(&old_path, &new_path).unwrap_err(), Error::Sys(Errno::ENOENT));
+}
+
+#[test]
+fn test_renameat() {
+    let old_dir = tempfile::tempdir().unwrap();
+    let old_dirfd = open(old_dir.path(), OFlag::empty(), Mode::empty()).unwrap();
+    File::create(&old_dir.path().join("old")).unwrap();
+    let new_dir = tempfile::tempdir().unwrap();
+    let new_dirfd = open(new_dir.path(), OFlag::empty(), Mode::empty()).unwrap();
+    renameat(old_dirfd, "old", new_dirfd, "new").unwrap();
+    assert_eq!(renameat(old_dirfd, "old", new_dirfd, "new").unwrap_err(),
+               Error::Sys(Errno::ENOENT));
+    close(old_dirfd).unwrap();
```

Comment:
How about a check that new_dir/"new" exists?

Reply:
Oops. Done.